express-notes-api/server.js

```diff
@@ -0,0 +1,157 @@
+import express from 'express';
+import { writeFile, readFile } from 'node:fs/promises';
+
+// --- Reusable functions ---
+function handleError(res, err) {
+  console.error('/api/notes error', err);
+  res.status(500).send({ error: 'An unexpected error occurred.' });
+}
+
+function evaluateId(res, journal, id) {
+  const validIds = Object.keys(journal.notes);
+  const errorMsg = `${id} is not a valid ID. Please provide a valid ID: ${validIds.join(', ')}`;
+  if (isNaN(id) || Number(id) < 0) {
+    res.status(400).send(errorMsg);
+  } else if (!validIds.includes(id)) {
+    res.status(404).send(errorMsg);
+  } else {
+    return true;
+  }
+}
+
+function evaluateReqBody(req, res, action) {
+  let userInputs = Object.keys(req.body);
+  userInputs = userInputs.filter((key) => key !== 'id');
+  switch (action) {
+    case 'delete':
```

Comment:
I agree with Tim's comment. With all the error checking in this one function, you now have to switch on the action, which means if you add a new action you have to modify this function in addition to the new action handler you write.

In general, you want to organize your code so that if you make changes to it (such as adding a new endpoint) you only have to change the code in one place. That's not always achievable, but it's the goal you should strive for.